client/dashboard/app/queries/domain-nameservers.ts

```diff
@@ -0,0 +1,44 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { fetchDomainNameservers, updateDomainNameservers } from '../../data/domain-nameservers';
+
+const useDomainNameserversQuery = ( domainName: string ) =>
+	useQuery( {
+		queryKey: [ 'domains', domainName, 'nameservers' ],
+		queryFn: () => fetchDomainNameservers( domainName ),
+		refetchOnWindowFocus: false,
+	} );
+
+interface UpdateNameserversOptions {
+	onSuccess?: ( nameservers: string[] ) => void;
+	onError?: ( error: Error ) => void;
+}
+
+function useUpdateNameserversMutation(
+	domainName: string,
+	queryOptions: UpdateNameserversOptions = {}
+) {
+	const queryClient = useQueryClient();
+	const mutation = useMutation( {
+		mutationFn: ( { nameservers }: { nameservers: string[] } ) =>
+			updateDomainNameservers( domainName, nameservers ),
+		...queryOptions,
+		onSuccess( nameservers ) {
+			queryClient.invalidateQueries( {
+				queryKey: [ 'domains', domainName, 'nameservers' ],
+			} );
+			queryOptions.onSuccess?.( nameservers );
```

Comment:
Continuing the comment about this layer only exporting the query/mutation options. The nice thing about the existing pattern is that we don't need to call additional `onSuccess()` like this manually. Instead, we can:

1. Specify `onSuccess` on the mutation options, e.g. invalidating queries:

https://github.com/Automattic/wp-calypso/blob/823ad556d8225db87a8398aba3e1d5824313949d/client/dashboard/app/queries/site-settings.ts#L16-L26

2. Call additional `onSuccess` when we call `mutate()`, e.g. showing notices:

https://github.com/Automattic/wp-calypso/blob/823ad556d8225db87a8398aba3e1d5824313949d/client/dashboard/sites/settings-agency/index.tsx#L94-L104